Fix newline bug in date-filter

ashsearle opened the pull request:
Formatting a date across 2 lines fails:
```html
<span>{{ '2017-03-09' | date: 'MMM\nyyyy' }}</span>
```

**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Bug fix


**What is the current behavior? (You can also link to an open issue here)**
The date format is only processed up to the newline.  i.e. as though the sample code was:

```html
<span>{{ '2017-03-09' | date: 'MMM' }}</span>
```

Producing:
```html
<span>Mar</span>
```

**What is the new behavior (if this is a feature change)?**
The full date format string is processed, producing:
```html
<span>Mar
2017</span>
```

**Does this PR introduce a breaking change?**
No.


**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit-message-format
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**Other information**:

ashsearle committed:
fix(date filter): handle newlines in format string

Closes #15794

nahidshahin reviewed (approved):
Solves the problem of '.' at regular expression that was excluding newline character '\n'. [\s\S] is a safe replacement.

HLJMT commented:
？？

gkalpak commented on test/ng/filter/filtersSpec.js:
```diff
@@ -504,6 +504,11 @@ describe('filters', function() {
 
+    it('should allow newlines in format', function() {
+      expect(date(midnight, 'EEE\nMMM d\'\n\'yy/xxx\n')).
+                      toEqual('Fri\nSep 3\n10/xxx\n');
```
Nit: There is not need for this to be a new line.

gkalpak reviewed (changes requested):
Generally LGTM.
Can you please update the commit message to follow our [guidelines][1]? That helps when generating the changelog.

[1]: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit

ashsearle committed:
fix(date filter): handle newlines in format string

Closes #15794

ashsearle commented:
I've updated commit message, addressed code review comment, and added short sentence to documentation as discussed in the issue #15794 - I hope the wording's OK.
